flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -185,6 +196,12 @@ public YarnResourceManager(
 		this.resource = Resource.newInstance(defaultTaskManagerMemoryMB, defaultCpus);
 
 		this.slotsPerWorker = createWorkerSlotProfiles(flinkConfig);
+
+		final int containerLauncherNum = flinkConfig.getInteger(YarnConfigOptions.CONTAINER_LAUNCHER_NUM_MAX);
+		this.startContainerExecutor = new ThreadPoolExecutor(containerLauncherNum, containerLauncherNum,
```

Comment:
Easier to use `Executors.newFixedThreadPool(size, ExecutorThreadFactory)`

Reply:
I think it would also be better to pass the `ExecutorService` into the `YarnResourceManager` using dependency injection.

Reply:
+1. according to the [Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html): `ExecutorService` can "provides methods to manage termination and methods that can produce a {@link Future} for tracking progress of one or more asynchronous tasks." seems to be a better fit for the purpose.

Reply:
also, I am considering, should we enforce a hard limit on how many threads a user can configure instead of a MAX_INT? currently the `maximumPoolSize` argument is set equivalent to `corePoolSize`.